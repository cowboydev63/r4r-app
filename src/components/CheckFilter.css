
.filter__label.r4r-DEFAULT {
    display: flex;
}

.checkbox__outline {
    display: inline-block;
    position: relative;
    margin-right: 5px;
    width: 15px;
    height: 15px;
    border: 2px solid #186d89;
    color: #186d89;
}

.checkbox__outline .checkbox__checkmark {
    position: absolute;
    height: 100%;
    width: 100%;
    top: -1px;
    left: -1px;
}

.filter__checkbox.r4r-DEFAULT {
   opacity: 0;
   position: absolute;
   z-index: -1;
}
