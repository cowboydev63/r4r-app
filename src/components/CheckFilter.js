import React from 'react';
import PropTypes from 'prop-types';

class CheckFilter extends React.PureComponent {
    static propTypes = {
        label: PropTypes.string.isRequired,
        checked: PropTypes.bool.isRequired,
        count: PropTypes.string.isRequired,
        className: PropTypes.string.isRequired,
        onChange: PropTypes.func.isRequired,
    }

    static defaultProps = {
        className: 'filter-box__filter',
        onChange: () => {}
    }

    // TODO: Massive work connecting all these filters
    onChange = () => {
        this.props.onChange(this.props.value);
    }

    render() {
        return (
            <label
                className={ this.props.className }
            >
                <input 
                    type="checkbox"
                    value={ this.props.label }
                    checked={ this.props.checked }
                    onChange={ this.onChange }
                    className='filter__checkbox'
                />
                <span className='filter__label'>{`${this.props.label} (${this.props.count})`}</span>
            </label>
        )
    }
}

export default CheckFilter;