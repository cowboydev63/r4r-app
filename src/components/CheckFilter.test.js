import React from 'react';
import { shallow } from 'enzyme';
import CheckFilter from './CheckFilter';

describe('CheckFilter Component', () => {
  it('renders without crashing', () => {
    shallow(<CheckFilter />);
  });

  it('renders correctly', () => {
    const wrapper = shallow(<CheckFilter />);
    expect(wrapper).toMatchSnapshot();
  })
})