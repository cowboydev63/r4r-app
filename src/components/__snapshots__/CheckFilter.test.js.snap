// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckFilter Component renders correctly 1`] = `
<Theme
  aria-checked={false}
  className="filter-box__filter"
  element="label"
  onKeyPress={[Function]}
  role="checkbox"
  tabIndex="0"
>
  <Theme
    checked={false}
    className="filter__checkbox"
    element="input"
    onChange={[Function]}
    tabIndex="-1"
    type="checkbox"
    value=""
  />
  <Theme
    className="filter__label"
    element="span"
  >
    <div
      className="checkbox__outline"
    />
     (0)
  </Theme>
</Theme>
`;
